src/Background/stat_estimators.jl

```diff
@@ -1,5 +1,43 @@
 using Statistics
 
+"""
+    sigma_clip(data, sigma_low, sigma_high; center=median, std=std, iterations=5)
+    sigma_clip(data, sigma; center=median, std=std, iteration=5)
+
+This function returns sigma clipped values of the input `data`.
+`sigma_high` and `sigma_low` are for un-symmetrical clipping, when `sigma_low = sigma_high` then they can be passed as `sigma`.
+`center`, `std` and `iterations` are optional parameters which are functions for finding central element, standard deviation and number of iterations of sigma
+clipping to be carried out.
+
+# Example
+```jldoctest
+julia> data = [1, 2, 3];
+
+julia> sigma_clip(data, 1)
+3-element Array{Float64,1}:
+ 1.0
+ 2.0
+ 3.0
+
+julia> sigma_clip(data, 1, 1)
+3-element Array{Float64,1}:
+ 1.0
+ 2.0
+ 3.0
+```
+"""
+
+function sigma_clip(data::AbstractArray, sigma_low::Number, sigma_high::Number; iterations::Number=5, center=median, std=std)
```

Comment:
I don't know what's Miles' plan here, whether to modify in place the data or not.  Note that in this case you aren't actually modifying `data`.

I'd maybe suggest to have both `sigma_clip!` and `sigma_clip`, with the former modifying in-place, the latter not, but based on the first one.  It could simply be something like:
```julia
sigma_clip(data, rest...; kwargs...) = sigma_clip!(copy(data), rest...; kwargs...)
```

@mileslucas what do you think?

Unrelated to the comment above, I'd restrict `sigma_low` and `sigma_high` to `Real` and `iterations` to `Integer` :wink:

Reply:
I have added both `sigma_clip!` and `sigma_clip`, and also modified the type restrictions for the suggested variables. :smiley: